index.js

```diff
@@ -56,42 +59,19 @@ Client.encoding = Object.freeze({
 function Client (opts) {
   if (!(this instanceof Client)) return new Client(opts)
 
-  this.config(opts)
-
-  Writable.call(this, this._conf)
-
-  const errorproxy = (err) => {
-    if (this.destroyed === false) this.emit('request-error', err)
-  }
-
-  const fail = () => {
-    if (this._writableState.ending === false) this.destroy()
-  }
+  Writable.call(this, { objectMode: true })
 
   this._corkTimer = null
   this._received = 0 // number of events given to the client for reporting
   this.sent = 0 // number of events written to the socket
+  this._agent = null
   this._active = false
   this._onflushed = null
   this._transport = null
   this._configTimer = null
   this._encodedMetadata = null
 
-  switch (this._conf.serverUrl.protocol.slice(0, -1)) { // 'http:' => 'http'
-    case 'http': {
-      this._transport = require('http')
-      break
-    }
-    case 'https': {
-      this._transport = require('https')
-      break
-    }
-    default: {
-      throw new Error('Unknown protocol ' + this._conf.serverUrl.protocol.slice(0, -1))
-    }
-  }
-
-  this._agent = new this._transport.Agent(this._conf)
+  this.config(opts)
```

Comment:
Reviewer note: `this._transport` and `this._agent` are now set in `this.config()`.